trade-finance-manager-api/api-tests/api-ssrf.api-test.js

```diff
@@ -0,0 +1,1349 @@
+const axios = require('axios');
+// eslint-disable-next-line import/no-extraneous-dependencies
+const MockAdapter = require('axios-mock-adapter');
+
+const api = jest.requireActual('../src/v1/api');
+
+const mockAxios = new MockAdapter(axios);
+
+describe('API is protected against SSRF attacks', () => {
+  describe('findOnePortalDeal', () => {
+    const mockResponse = {
+      deal: 'Mock deal',
+    };
+    beforeAll(() => {
+      mockAxios.reset();
+      const url = /^.*\/v1\/portal\/deals\/.*$/;
+      mockAxios.onGet(url).reply(200, mockResponse);
+    });
+    it('Returns an error when a url traversal is supplied', async () => {
+      const urlTraversal = '../../../etc/stealpassword';
+      const expectedResponse = { status: 400, data: 'Invalid deal id' };
+
+      const response = await api.findOnePortalDeal(urlTraversal);
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Returns an error when a local IP is supplied', async () => {
+      const localIp = '127.0.0.1';
+      const expectedResponse = { status: 400, data: 'Invalid deal id' };
+
+      const response = await api.findOnePortalDeal(localIp);
+
+      expect(response).toMatchObject(expectedResponse);
+    });
+
+    it('Makes an axios request when the deal id is valid', async () => {
+      const validDealId = '5ce819935e539c343f141ece';
```

Comment:
Can we add assertions where deal ID is in valid format but does not exists:

Reply:
Same applies to all of the below please.

Reply:
As I'm completely mocking responses here is there much we gain from having a valid deal ID but doesn't exist? I'd just be changing the mockResponse in `mockAxios.onGet(url).reply(200, mockResponse);` and then asserting that it equals the mock response later on so it feels like the same test as having a valid ID

Reply:
True, happy to leave it as it is thank you for the explanation.